vita3k/kernel/src/thread.cpp

```diff
@@ -156,52 +156,79 @@ int ThreadState::start(KernelState &kernel, SceSize arglen, const Ptr<void> &arg
     }
     something_to_do.notify_one();
 
-    if (kernel.thread_event_start) {
-        int ret = run_callback(kernel.thread_event_start.address(), { SCE_KERNEL_THREAD_EVENT_TYPE_START, static_cast<uint32_t>(id), 0, kernel.thread_event_start_arg });
-        if (ret != 0)
-            LOG_WARN("Thread start event handler returned {}", log_hex(ret));
-    }
-
     return SCE_KERNEL_OK;
 }
 
-void ThreadState::exit(KernelState &kernel, SceInt32 status) {
+void ThreadState::exit(SceInt32 status) {
     if (kernel.thread_event_end) {
         int ret = run_callback(kernel.thread_event_end.address(), { SCE_KERNEL_THREAD_EVENT_TYPE_END, static_cast<uint32_t>(id), 0, kernel.thread_event_end_arg });
         if (ret != 0)
             LOG_WARN("Thread end event handler returned {}", log_hex(ret));
     }
 
     std::lock_guard<std::mutex> guard(mutex);
+    run_end_callback = true;
```

Comment:
We already run end callback. Why run twice?

Reply:
Sorry forgot to delete the run_callback above.